data_collator.py

```diff
@@ -0,0 +1,114 @@
+
+        batch["input_ids"], batch["labels"] = self.mask_tokens(
+            batch["input_ids"], special_tokens_mask=special_tokens_mask
+        )
+
+        return batch
+
+    def mask_tokens(self,
+                    inputs: jnp.ndarray,
+                    special_tokens_mask: Optional[jnp.ndarray] = None
+                    ) -> Tuple[jnp.ndarray, jnp.ndarray]:
+
+        inputs_copy = np.array(inputs)
+        labels = np.array(inputs)
+        if special_tokens_mask is None:
+            special_tokens_mask = [
+                self.tokenizer.get_special_tokens_mask(val, already_has_special_tokens=True) for val in labels.tolist()
+            ]
+            special_tokens_mask = jnp.array(special_tokens_mask, dtype=bool)
+        else:
+            special_tokens_mask = special_tokens_mask.bool()
+
+        # determine how many tokens we need to mask in total
+        is_token = ~(labels == self.tokenizer.pad_token_id) & ~special_tokens_mask
+        num_to_mask = int(math.ceil(is_token.astype(float).sum() * self.mlm_probability))
+        if num_to_mask == 0:
+            return inputs, labels
+
+        # generate a sufficient number of span lengths
+        #poisson_distribution = poisson(lam=self.poisson_lambda,size=(num_to_mask,))
+        lengths = poisson(lam=self.poisson_lambda,size=(num_to_mask,))
+        while np.cumsum(lengths, 0)[-1] < num_to_mask:
+            lengths = np.concatenate([lengths, poisson(rate=self.poisson_lambda,size=(num_to_mask,))])
+
+        # remove all spans of length 0
+        # Note that BART inserts additional mask tokens where length == 0,
+        # which we do not implement for now as it adds additional complexity
+        lengths = lengths[lengths > 0]
+
+        # trim to about num_to_mask tokens
+        idx = np.argmin(np.abs(np.cumsum(lengths, 0) - num_to_mask)) + 1
+        lengths = lengths[:idx + 1]
+
+        # select span start indices
+        token_indices = jnp.argwhere(is_token==0)
+        span_starts = permutation(token_indices.shape[0])[:lengths.shape[0]]
+
+        # prepare mask
+        masked_indices = np.array(token_indices[span_starts])
+        mask = np.full_like(labels, fill_value=False)
+
+        # mask span start indices
+        for mi in masked_indices:
+            mask[tuple(mi)] = True
+        lengths -= 1
+
+        # fill up spans
+        max_index = labels.shape[1] - 1
+        remaining = (lengths > 0) & (masked_indices[:, 1] < max_index)
+        while np.any(remaining):
+            masked_indices[remaining, 1] += 1
+            for mi in masked_indices:
+                mask[tuple(mi)] = True
+            lengths -= 1
+            remaining = (lengths > 0) & (masked_indices[:, 1] < max_index)
+
+        # place the mask tokens
+        mask[np.where(special_tokens_mask==True)] = False
+        inputs_copy[np.where(mask==1)] = self.tokenizer.mask_token_id
+        labels[np.where(mask == 0)] = -100
```

Comment:
Looking at the labels output, they're all -100, guessing something here needs to be fixed

```
[[-100 -100 -100 ... -100 -100 -100]
 [-100 -100 -100 ... -100 -100 -100]
 [-100 -100 -100 ... -100 -100 -100]]
```

Reply:
Resolved